tests/test_services/test_global_parameters_service.py

```diff
@@ -0,0 +1,142 @@
+"""Tests for Global Config Service."""
+import os
+from stat import S_IREAD
+from unittest import mock
+
+import pytest
+import yaml
+
+from matcha_ml.errors import MatchaPermissionError
+from matcha_ml.services.global_parameters_service import GlobalParameters
+
+INTERNAL_FUNCTION_STUB = "matcha_ml.services.global_parameters_service.GlobalParameters"
+
+
+@pytest.fixture(autouse=True)
+def teardown_singleton():
+    """Tears down the singleton before each test case by clearing the current object."""
+    GlobalParameters._instance = None
+
+
+def test_class_is_singleton(matcha_testing_directory):
+    """Tests that the GlobalParameters is correctly implemented as a singleton.
+
+    Args:
+        matcha_testing_directory (str): Mock testing directory location for the GlobalConfig file to be located
+    """
+    with mock.patch(
+        f"{INTERNAL_FUNCTION_STUB}.default_config_file_path",
+        new_callable=mock.PropertyMock,
+    ) as file_path:
+        file_path.return_value = str(
+            os.path.join(str(matcha_testing_directory), ".matcha-ml", "config.yaml")
+        )
+
+        first_instance = GlobalParameters()
+        second_instance = GlobalParameters()
+        assert first_instance is second_instance
```

Comment:
Nit: even though `is` checks whether two references refer to the same object, for completeness it'd be good to add in checks for `user_id` and the analytics bool